google/cloud/internal/open_telemetry_enabled_test.cc

```diff
@@ -0,0 +1,23 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+
+TEST(OpenTelemetry, VerifyEnabled) {
+#ifdef GOOGLE_CLOUD_CPP_HAVE_OPEN_TELEMETRY
```

Comment:
Feels that a better test would be something like:

```cc
#ifdef GOOGLE_CLOUD_CPP_HAVE_OPEN_TELEMETRY
#include <opentelemetry/version.h>
#include <gmock/gmock.h>

TEST(OpenTelemetry, VerifyUsable) {
  auto version = std::string{OPENTELEMETRY_VERSION};
  EXPECT_THAT(version, Not(IsEmpty());
  ... ... Maybe use a function here ... ... 
}
```

Reply:
I am testing a different thing. I want a test that verifies that `GOOGLE_CLOUD_CPP_HAVE_OPEN_TELEMETRY` is defined. If it was not defined than the thing you proposed is an empty test fixture, which will quietly succeed.

We should have a test like the one you propose. But we are not ready for it yet.

Reply:
But that is an odd test.  You are basically testing that CMake / Bazel works as advertised?

Reply:
It is a weird test. But it is the only thing that will notice if we accidentally remove the `defines` / `target_compile_definitions`